### DISCUSSION TITLE
Rename master realm name in KC17

### DISCUSSION THREAD

**Discussion by gjain-sag** [keycloak/keycloak#11027] [Archive]
In Jboss based distribution we used to update this field to rename master realm name to something custom:
           
" <master-realm-name>mymaster</master-realm-name>"

How can the same be achieved for quarkus based distribution. Is it supported ?

**Comment by ajaiswal595** [comment]
@pedroigor I did not find any solution on the internet.

Is there any way to replace master-realm with My-Realm?

We used to do in standalone.xml in wildfly distro.

<subsystem xmlns="urn:jboss:domain:keycloak-server:1.1">
            <web-context>auth</web-context>
            <providers>
                ...
            </providers>
            <master-realm-name>MyRealm</master-realm-name>

**Reply by gjain-sag** [reply]
You can set this as your environment variable to set Default Master Realm Name:
SET KC_SPI_ADMIN_REALM=<Master Realm Name>

**Reply by pedroigor** [reply]
@ajaiswal595 @gjain-sag Thanks, that is the option you should set. 

Please, see https://www.keycloak.org/server/configuration-provider.

**Reply by ajaiswal595** [reply]
Thanks, @gjain-sag and @pedroigor  KC_SPI_ADMIN_REALM=MyRealm worked and master-realm was replaced with MyRealm.
